pkgs/by-name/oc/ocis/package.nix

```diff
@@ -0,0 +1,87 @@
+{
+  lib,
+  stdenvNoCC,
+  fetchFromGitHub,
+  buildGoModule,
+  callPackage,
+  gnumake,
+  pnpm,
+  nodejs,
+  ocis,
+}:
+let
+  idp-assets = stdenvNoCC.mkDerivation {
+    pname = "idp-assets";
+    version = "0-unstable-2020-10-14";
+    src = fetchFromGitHub {
+      owner = "owncloud";
+      repo = "assets";
+      rev = "e8b6aeadbcee1865b9df682e9bd78083842d2b5c";
+      hash = "sha256-PzGff2Zx8xmvPYQa4lS4yz2h+y/lerKvUZkYI7XvAUw=";
+    };
+    installPhase = ''
+      mkdir -p $out/share
+      cp logo.svg favicon.ico $out/share/
+    '';
+    dontConfigure = true;
+    dontBuild = true;
+    dontFixup = true;
+  };
+in
+buildGoModule rec {
+  pname = "ocis";
+  version = "5.0.9";
+
+  vendorHash = null;
+
+  src = fetchFromGitHub {
+    owner = "owncloud";
+    repo = "ocis";
+    rev = "refs/tags/v${version}";
+    hash = "sha256-TsMrQx+P1F2t66e0tGG0VvRi4W7+pCpDHd0aNsacOsI=";
+  };
+
+  nativeBuildInputs = [
+    gnumake
+    nodejs
+    pnpm.configHook
+  ];
+
+  pnpmDeps = pnpm.fetchDeps {
+    inherit pname version src;
+    sourceRoot = "${src.name}/services/idp";
+    hash = "sha256-gNlN+u/bobnTsXrsOmkDcWs67D/trH3inT5AVQs3Brs=";
+  };
+  pnpmRoot = "services/idp";
+
+  buildPhase = ''
+    runHook preBuild
+    cp -r ${ocis.web}/share/* services/web/assets/
+    pnpm -C services/idp build
+
+    mkdir -p services/idp/assets/identifier/static
+    cp -r ${idp-assets}/share/* services/idp/assets/identifier/static/
+
+    make -C ocis VERSION=${version} DATE=${version} build
+    runHook postBuild
+  '';
+
+  installPhase = ''
+    runHook preInstall
+    mkdir -p $out/bin/
+    cp ocis/bin/ocis $out/bin/
+    runHook postInstall
+  '';
+
+  passthru = {
+    web = callPackage ./web.nix { };
```

Comment:
```suggestion
    web = ocis.web;
```

Reply:
@FliegendeWurst Sorry but I'm not familiar with this usage. I encountered infinite recursion with this suggested change applied. Should I make other changes in order to make this work?

Reply:
Hmm.. I suppose you could just omit the passthru.web. Unless you really need it somewhere.

Reply:
Well, it's directly used by the code [here](https://github.com/NixOS/nixpkgs/blob/cc960b6eeeaf40964ec547ff849293a74cd4f367/pkgs/by-name/oc/ocis/package.nix#L59). Plus, it's a separate frontend that could theoretically be hosted independently, so I don't think keeping `passthru.web` does any harm..?

Reply:
In that case it is okay.

Reply:
Pushed one more time for the mainProgram change. Thanks for review!